src/devices/BrickPi3/SPIExceptions.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Iot.Device.BrickPi3
+{
+
+    /// <summary>
+    /// Exceptions classes
+    /// </summary>
+    public class SpiExceptions
```

Comment:
Given that none of these really provide any extra information to a regular exception, I would just instead throw .NET exceptions based on where they are needed. For example: IOException, ArgumentException, InvalidOperationException, etc. I don't think it is worth it to educate consumers on new exception types that don't really provide anything extra but the name.

Reply:
Good point. I will remove all. IOException will be enough.